programs/fixed-term/src/margin/instructions/configure_auto_roll.rs

```diff
@@ -0,0 +1,56 @@
+use anchor_lang::prelude::*;
+use jet_program_common::Fp32;
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::FromPrimitive;
+
+use crate::{
+    margin::state::{AutoRollConfig, MarginUser},
+    FixedTermErrorCode,
+};
+
+#[derive(FromPrimitive, ToPrimitive, Clone, Copy, AnchorDeserialize, AnchorSerialize)]
+#[repr(u8)]
+pub enum MarketSide {
+    Borrowing,
+    Lending,
+}
+
+#[derive(Accounts)]
+pub struct ConfigureAutoRoll<'info> {
+    /// The `MarginUser` account.
+    /// This account is specific to a particular fixed-term market
+    #[account(
+        mut,
+        has_one = margin_account,
+    )]
+    pub margin_user: Box<Account<'info, MarginUser>>,
+
+    /// The signing authority for this user account
+    #[account(
+        constraint = margin_account.owner == &jet_margin::ID,
+    )]
+    pub margin_account: Signer<'info>,
```

Comment:
this is the wrong way to validate a margin account

```suggestion
    #[account(signer)]
    pub margin_account: AccountLoader<'info, MarginAccount>,
```